src/sentry/api/endpoints/organization_repository_details.py

```diff
@@ -49,13 +50,21 @@ def put(self, request, organization, repo_id):
 
         serializer = RepositorySerializer(data=request.DATA, partial=True)
 
-        if serializer.is_valid():
-            result = serializer.object
-            if result.get('status'):
-                if result['status'] in ('visible', 'active'):
-                    repo.update(status=ObjectStatus.VISIBLE)
-                else:
-                    raise NotImplementedError
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        result = serializer.object
+        update_kwargs = {}
+        if result.get('status'):
+            if result['status'] in ('visible', 'active'):
+                update_kwargs['status'] = ObjectStatus.VISIBLE
+            else:
+                raise NotImplementedError
+        if result.get('integrationId'):
+            update_kwargs['integration_id'] = result['integrationId']
```

Comment:
i just realized i need to add validation that the integration exists / should check that it is linked to org.

also @MeredithAnya pointed out that we also need to update the repo provider

gonna fix both of those things (and will add some more tests)